### PR TITLE
Admin login + product list

### PR DESCRIPTION
**Pull request by oliverheywood451**
# Description

First off, sorry this is so large. Lots of file renames and stuff copied into the seller folder.

- In the buyer, renamed OcFormError, OcMatchField, OcQuantityError to avoid confusion with the SDK.
- Seller - Adding login functionality. Adding basic Routes. Adding Product list. 
- Would love feedback on the new sorting component. It something that might get used a lot in Seller.

## Type of change

- [ x] New feature (non-breaking change which adds functionality)
- [x ] Refactor (change/optimize code without changing external behavior)

# Checklist:

- [ x] I have performed a self-review of my own code
- [ x] I have commented my code, particularly in hard-to-understand areas
- [ x] I have added tests that prove my fix is effective or that my feature works
- [ x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/UI/Seller/src/app/shared/components/sort-table-header/sort-table-header.component.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  styleUrls: ['./sort-table-header.component.scss'],
+})
+export class SortTableHeaderComponent {
+  constructor() {}
+  faCaretUp = faCaretUp;
+  faCaretDown = faCaretDown;
+
+  // The text to display to the user on the header for this column
+  @Input() displayName: string;
+
```
Comment (crhistianramirez) [inline src/UI/Seller/src/app/shared/components/sort-table-header/sort-table-header.component.ts]: A better name might be `shared-sort-column`, i was a little confused when i ran into this in the html first and looked like an array of tables inside a table tag ;)

Comment (crhistianramirez) [reply src/UI/Seller/src/app/shared/components/sort-table-header/sort-table-header.component.ts]: and actually maybe just call it column? Looks like sort is optional.

Comment (oliverheywood451) [reply src/UI/Seller/src/app/shared/components/sort-table-header/sort-table-header.component.ts]: My intention is that if you use this component you always get the ability to sort by that column.

Comment (oliverheywood451) [reply src/UI/Seller/src/app/shared/components/sort-table-header/sort-table-header.component.ts]: I like 'shared-sort-column'

**Pull request merged**